books/u-physics/bake

```diff
@@ -0,0 +1,134 @@
+    # In-place bake exercises & solutions
+    chapter.search('.review-conceptual-questions, .review-problems, .review-additional-problems, .review-challenge').exercises.each do |exercise|
+      BakeNumberedExercise.v1(exercise: exercise, number: exercise.count_in(:chapter))
+    end
+
+    # Bake answer key from chapter/ move solutions from eoc into answer key
+    MoveSolutionsToAnswerKey.v1(chapter: chapter, metadata_source: book.metadata, strategy: :uphysics, append_to: solutions_container)
+  end
+
+  book.search('[data-type="solution"]').each do |solution|
+    # Add the 'has-first-element' class to elements that need it
+    BakeFirstElements.v1(within: solution,
+                         selectors: ['.os-solution-container > [data-type="media"]',
+                                     '.os-solution-container > .os-table'])
+  end
+
+  book.chapters.each do |chapter|
+    chapter.tables('$:not(.unnumbered)').each do |table|
+      BakeNumberedTable.v1(table: table,
+                           number: "#{chapter.count_in(:book)}.#{table.count_in(:chapter)}")
+    end
+
+    chapter.examples.each do |example|
+      BakeExample.v1(example: example,
+                     number: "#{chapter.count_in(:book)}.#{example.count_in(:chapter)}",
+                     title_tag: 'h3')
+    end
+
+    chapter.figures('$:not(.unnumbered)').each do |figure|
+      BakeFigure.v1(figure: figure,
+                    number: "#{chapter.count_in(:book)}.#{figure.count_in(:chapter)}")
+    end
+  end
+
+  book.pages('$.appendix').each do |page|
+    appendix_letter = [*('A'..'Z')][page.count_in(:book) - 1]
+    BakeAppendix.v1(page: page, number: appendix_letter)
+
+    page.tables('$:not(.unnumbered)').each do |table|
+      BakeNumberedTable.v1(table: table, number: "#{appendix_letter}#{table.count_in(:page)}")
+    end
+  end
+
+  BakeUnnumberedTables.v1(book: book)
+  BakeEquations.v1(book: book)
+  BakeMathInParagraph.v1(book: book)
+
+  # Fix titles in chapter review composite pages
+  # TODO: remove this
+  book.chapters('.os-eoc.os-chapter-review-container').composite_pages.each do |page|
+    title = page.search('[data-type="document-title"]').first
+    title[:'data-type'] = 'title'
+    title.name = 'h3'
+    page.metadatas.first.search('[data-type="document-title"]').first.replace_children(with: 'Chapter Review')
+  end
+
+  BakeIndex.v1(book: book)
+  BakeCompositePages.v1(book: book)
+  BakeCompositeChapters.v1(book: book)
+  BakeFootnotes.v1(book: book)
+  BakeToc.v1(book: book)
+  BakeLinkPlaceholders.v1(book: book)
+  BakeUnitTitle.v1(book: book)
+
+end
+
+after = Kitchen::BookRecipe.new(book_short_name: :u_physics) do |doc|
+  include Kitchen::Directions
+
+  book = doc.book
+
+  # remove w3 from toc
+  # TODO: remove this
+  book.search('nav#toc').first.search('[xmlns="http://www.w3.org/1999/xhtml"]').each do |el|
+    el.remove_attribute('xmlns')
+  end
```

Comment:
will be addressed by openstax/kitchen#209

Reply:
this has been merged